sockeye/layers.py

```diff
@@ -225,6 +228,85 @@ def __call__(self,
                                     bias=bias,
                                     flatten=False)
 
+class LengthRatioConfig(config.Config):
+    """
+    Configuration of the length ratio predictor.
+
+    :param layers: Number of layers.
+    :param weight: Weight of this loss.
+    """
+
+    def __init__(self, num_layers: int, weight: float) -> None:
+        super().__init__()
+        self.num_layers = num_layers
+        # TODO: keep weight here redundant because the weight is also stored in the loss config.
+        # but it's used to test if we need length prediction at all
+        self.weight = weight
+
+
+class LengthRatio:
+    """
+    Defines the length-ratio prediction layer of Sockeye.
+
+    :param hidden_size: Decoder hidden size.
+    :param prefix: Prefix used for naming.
+    """
+
+    def __init__(self,
+                 hidden_size: int,
+                 num_layers: int,
+                 weight_normalization: bool,
```

Comment:
does this really need weight normalization?

Reply:
Good question, I guess we don't know yet, but removed for now.